servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/FromInputStreamPublisherTest.java

```diff
@@ -368,37 +368,77 @@ void readsAllBytesWhenAvailableNotImplemented() throws Throwable {
     }
 
     @ParameterizedTest(name = "{displayName} [{index}] readChunkSize={0}")
-    @ValueSource(ints = {7, 1024})
+    @ValueSource(ints = {4, 1024})
     void doNotFailOnInputStreamWithBrokenAvailableCall(int readChunkSize) throws Throwable {
-        initChunkedStream(bigBuff, of(5, 0, 0, 10, 5, 5, 1, 0),
-                                   of(5, 7, 7, 10, 5, 5, 1, 0));
+        // We use double "0, 0" because FromInputStreamPublisher does two calls to available() now. For this test, both
+        // calls to "broken" available() should consistently return `0`.
+        initChunkedStream(bigBuff, of(3, 0, 0, 4, 0, 0, 5, 0, 0, 2, 0, 0,    0, 0,       4, 0),
+                                   of(3, 7,    4, 4,    5, 2, 2, 2, 1, 2, 1, 1, 1, 1, 1, 4, 0));
         pub = new FromInputStreamPublisher(inputStream, readChunkSize);
 
         if (readChunkSize > bigBuff.length) {
             byte[][] items = {
-                    new byte[]{0, 1, 2, 3, 4},
+                    new byte[]{0, 1, 2},
                     // avail == 0 -> override to readChunkSize
-                    new byte[]{5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27,
-                            28, 29, 30, 31, 32, 33, 34, 35, 36},
+                    new byte[]{3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26,
+                            27, 28, 29, 30, 31, 32, 33, 34, 35, 36},
             };
             verifySuccess(items);
         } else {
             byte[][] items = {
-                    new byte[]{0, 1, 2, 3, 4},
-                    // avail == 0 -> override to readChunkSize
-                    new byte[]{5, 6, 7, 8, 9, 10, 11},
-                    // avail == 0 -> override to readChunkSize
-                    new byte[]{12, 13, 14, 15, 16, 17, 18},
-                    // readChunkSize < available
-                    new byte[]{19, 20, 21, 22, 23, 24, 25},
-                    new byte[]{26, 27, 28, 29, 30},
-                    new byte[]{31, 32, 33, 34, 35},
-                    new byte[]{36},
+                    // available < readChunkSize
+                    new byte[]{0, 1, 2},
```

Comment:
Changes for this test are made in the first commit to demonstrate that they only enhance test coverage and are not driven by the actual change. Consider reviewing this PR by every commit.